addons/l10n_ar_base/models/res_partner.py

```diff
@@ -0,0 +1,126 @@
+##############################################################################
+# For copyright and license notices, see __manifest__.py file in module root
+# directory
+##############################################################################
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.safe_eval import safe_eval
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    l10n_ar_cuit = fields.Char(
+        compute='_compute_l10n_ar_cuit',
+    )
+    l10n_ar_formated_cuit = fields.Char(
+        compute='_compute_l10n_ar_formated_cuit',
+    )
+    l10n_ar_id_number = fields.Char(
+        string='Identification Number',
+    )
+    l10n_ar_id_category_id = fields.Many2one(
+        string="Identification Category",
+        comodel_name='l10n_ar_id_category',
+        index=True,
+        auto_join=True,
+    )
+
+    @api.multi
+    def cuit_required(self):
+        """ Return the cuit number is this one is defined if not raise an
+        UserError
+        """
+        self.ensure_one()
+        if not self.l10n_ar_cuit:
+            raise UserError(_(
+                'No CUIT configured for partner [%i] %s') % (
+                    self.id, self.name))
+        return self.l10n_ar_cuit
```

Comment:
Is this not the same as a required field?  Also here, what if the partner is not related to Argentina localization?

Reply:
This method is called later by other modules that needs a CUIT to perform some action. For eg. create an electronic invoice, build some report, etc. The CUIT  is not always required on the partner, the partner can have different types of identifications, but for some specific operations you need the partner to be identified by a CUIT.

So, instead of making that validation on each method that requires a CUIT, we just call "cuit = partner.cuit_required()" and if the CUIT is not configured we raise the error to the user.

Reply:
let's call it ensure_cuit() and add comment on how to use it